async/logic/Async4BackendRest_CallbackAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,343 @@
+				continue;
+			}
+			
+			System.out.println("Async4BackendRest_CallbackAPI_Topic messages picked");
+			System.out.println("Test1");
+
+			typeresponse = ExecuteCallBackAPI(callbackUri, destinationStatusCode, destinationResponseBody, transactionid, asyncid, executiontime, destinationresult);
+			System.out.println("Test2");
+			if (((Response) typeresponse).getStatus() != Response.Status.CREATED.getStatusCode()) {
+				new AsyncResult<Response>(typeresponse);
+
+			}else{
+				
+			}
+			
+			
+			// 1秒待つ
+			try {
+				TimeUnit.SECONDS.sleep(1);
+				System.out.println("Test");
+				//return new AsyncResult<Response>(typeresponse);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+			
+		}
+		
+	}
+	
+	private Response PickupMessageForResponseQueue(StringBuilder destinationStatusCode, StringBuilder destinationResponseBody, 
+	StringBuilder callbackUri, StringBuilder destinationresult, StringBuilder transactionid, StringBuilder asyncid, 
+	StringBuilder executiontime) {
+	    Responses response      = new Responses();
+	    String receiveMessage = new String();
+	    
+	    destinationStatusCode.delete(0, destinationStatusCode.length());
+    	destinationResponseBody.delete(0, destinationResponseBody.length());
+    	callbackUri.delete(0, callbackUri.length());
+		destinationresult.delete(0, destinationresult.length());
+		transactionid.delete(0, transactionid.length());
+		asyncid.delete(0, asyncid.length());
+		executiontime.delete(0, executiontime.length());
+    	
+	    // 接続時の設定値を Properties インスタンスとして構築する
+	    Properties properties = new Properties();
+	    // 接続先 Kafka ノード
+	    properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "asynckafka1:9092");
+	    // Consumer を識別するための group id
+	    properties.put(ConsumerConfig.GROUP_ID_CONFIG, "java-consumer-group");
+	    // 読み取り位置である offset が未指定だった場合に、Kafka 上に最も早く置かれたメッセージを読み取る
+	    properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+	    // 読み取り後に、読み取り位置である offset を自動で更新する
+	    properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+	    // 最大1つのメッセージを取得する。
+	    properties.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1);
+	 
+	    // Consumer を構築する
+	    KafkaConsumer<String, String> consumer =
+	        new KafkaConsumer<String, String>(properties, new StringDeserializer(), new StringDeserializer());
+	 
+	    // Consumer をトピックに割り当てる
+	    consumer.subscribe(Arrays.asList("Async4BackendRest_CallbackAPI_Topic"));
+	 
+	    try {
+	      while (true) {
+	        // メッセージをとりだす
+	        ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(60l));
+	        Date date = new Date();
+    		SimpleDateFormat dateformat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    		String strDate = dateformat.format(date);
+	        
+	        // メッセージが無かったらそのまま抜ける
+			if(0 ==  records.count()) {
+				consumer.close();
+				return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource returned by the following method call: *build*. The resource is returned at line 159. There are other execution paths that don&#39;t close the resource or return it. Ensure the object returned by *build()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).